src/npctalk.cpp

```diff
@@ -2255,6 +2256,14 @@ void dialogue::gen_responses( const talk_topic &the_topic )
             // TODO: Allow NPCs to break training properly
             // Don't allow them to walk away in the middle of training
             std::stringstream reasons;
+            int part;
+            vehicle *veh = g->m.veh_at( p->pos(), part );
+            if( veh != nullptr ) {
+                if( abs( veh->velocity ) > 0 ) {
+                    reasons << _( "Let's not do it in the moving vehicle." ) << std::endl;
```

Comment:
This statement seems to convey something else, perhaps the message could be more along "I'm not going to train you while you're operating a vehicle!" or "I can't train you properly if we're in a moving vehicle!"